### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits): generalize universes in `Preserves.Filtered` and add `Reflects` classes

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Preserves/Filtered.lean [hunk 1]
```diff
@@ -48,14 +58,111 @@ instance compPreservesFilteredColimits (F : C ⥤ D) (G : D ⥤ E) [PreservesFil
...
+    [PreservesFilteredColimitsOfSize.{w', w} F] : PreservesFilteredColimitsOfSize.{0, 0} F :=
+  preservesFilteredColimitsOfSizeShrink F
+
+end Preserves
+
+section Reflects
+
+-- This should be used with explicit universe variables.
+/-- `ReflectsFilteredColimitsOfSize.{w', w} F` means that whenever the image of a filtered cocone
+under `F` is a colimit cocone, the originial cocone was already a colimit. -/
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Preserves/Filtered.lean:92]:
```suggestion
under `F` is a colimit cocone, the original cocone was already a colimit. -/
```

File: Mathlib/CategoryTheory/Limits/Preserves/Filtered.lean [hunk 2]
```diff
@@ -22,23 +20,35 @@ open CategoryTheory.Functor
...
-  preserves_filtered_colimits :
-    ∀ (J : Type v) [SmallCategory J] [IsFiltered J], PreservesColimitsOfShape J F
+abbrev PreservesFilteredColimits (F : C ⥤ D) : Type max u₁ u₂ (v + 1) :=
+  PreservesFilteredColimitsOfSize.{v, v} F
 
-attribute [instance 100] PreservesFilteredColimits.preserves_filtered_colimits
+attribute [instance 100] PreservesFilteredColimitsOfSize.preserves_filtered_colimits
 
 instance (priority := 100) PreservesColimits.preservesFilteredColimits (F : C ⥤ D)
     [PreservesColimits F] : PreservesFilteredColimits F where
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Preserves/Filtered.lean:54]: Should not this be generalized to a `...OfSize` version? (Same remark for `ReflectsColimits` below.)

File: Mathlib/CategoryTheory/Limits/Preserves/Filtered.lean [hunk 3]
```diff
@@ -22,47 +20,240 @@ open CategoryTheory.Functor
...
 
-attribute [instance 100] PreservesFilteredColimits.preserves_filtered_colimits
+attribute [instance 100] PreservesFilteredColimitsOfSize.preserves_filtered_colimits
 
 instance (priority := 100) PreservesColimits.preservesFilteredColimits (F : C ⥤ D)
-    [PreservesColimits F] : PreservesFilteredColimits F where
+    [PreservesColimitsOfSize.{w, w'} F] : PreservesFilteredColimitsOfSize.{w, w'} F where
   preserves_filtered_colimits _ := inferInstance
 
 instance compPreservesFilteredColimits (F : C ⥤ D) (G : D ⥤ E) [PreservesFilteredColimits F]
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Preserves/Filtered.lean:57]: This instance (and maybe a few other) should generalized to `...OfSize` versions.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**